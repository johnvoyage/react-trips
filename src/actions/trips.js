import uuid from 'uuid';

export const addTrip = (
  {
    country = ''
  } = {}
) => ({
  type: 'ADD_TRIP',
  trip: {
    id: uuid(),
    country,
  }
});

export const removeTrip = ({ id } = {}) => ({
  type: 'REMOVE_TRIP',
  id
});

export const editTrip = (id, updates) => ({
  type: 'EDIT_TRIP',
  id,
  updates
});