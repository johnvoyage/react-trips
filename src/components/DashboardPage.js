import React from 'react';

const DashboardPage = () => (
  <p>Dashboard page</p>
);

export default DashboardPage;