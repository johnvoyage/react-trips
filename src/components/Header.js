import React from 'react';
import { NavLink } from 'react-router-dom';


const Header = () => (
  <header>
    <h1>header</h1>
    <NavLink to="/" activeClassName='is-active' exact={true}>Home</NavLink>
    <NavLink to="/trips" activeClassName='is-active' exact={true}>Trips</NavLink>
    <NavLink to="/trips/new" activeClassName='is-active' exact={true}>New Trip</NavLink>

    <NavLink to="/profile" activeClassName='is-active'>Profile</NavLink>
  </header>
);
export default Header;