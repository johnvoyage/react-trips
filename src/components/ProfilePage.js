import React from 'react';

const ProfilePage = (props) => {
  console.log(props);
  return (
    <p>profile page for user: {props.match.params.id}</p>
  );
};

export default ProfilePage;