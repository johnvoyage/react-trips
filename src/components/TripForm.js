import React from 'react';
import moment from 'moment';
import { SingleDatePicker } from 'react-dates';


class TripForm extends React.Component {
  constructor(props) {
    super(props)

    this.state = {
      country: props.trip ? props.trip.country : '',
      note: props.trip ? props.trip.note :'',
      amount: '',
      createdAt: props.trip ? moment(props.trip.createdAt) : moment(),
      calendarFocused: false,
      error: '',
    };
  }

  onCountryChange = (event) => {
    const country = event.target.value;
    this.setState(() => ({ country }))
  };

  onNoteChange = (event) => {
    const note = event.target.value;
    this.setState(() => ({ note }))
  };

  onAmountChange = (event) => {
    const amount = event.target.value;
    
    if (!amount || amount.match(/^\d{1,}(\.\d{0,2})?$/)) {
      this.setState(() => ({ amount }))
    };
  };

  onDateChange = (createdAt) => {
    if (createdAt) {
      this.setState(() => ({ createdAt }))
    };
  };

  onFocusChange = ({ focused }) => {
    this.setState(() => ({ calendarFocused: focused }))
  }

  onSubmit = (event) => {
    event.preventDefault();

    if (!this.state.country) {
      this.setState(() => ({ error: "Please fill out the form" }));
    } else {
      this.setState(() => ({ error: '' }))
      this.props.onSubmit({
        country: this.state.country
      })
    };

  };

  render() {
    return (
      <div>
        {this.state.error && <p>{this.state.error}</p>}
        <form onSubmit={this.onSubmit}>
          <input 
            autoFocus
            type="text"
            placeholder="Country"
            value={this.state.country}
            onChange={this.onCountryChange}
          />
          <input 
            type="text"
            placeholder="number for regex prctce"
            value={this.state.amount}
            onChange={this.onAmountChange}
          />
          <SingleDatePicker
            date={this.state.createdAt}
            onDateChange={this.onDateChange}
            focused={this.state.calendarFocused}
            onFocusChange={this.onFocusChange}
          />
          <textarea 
            placeholder="Add a note..."
            onChange={this.onNoteChange}
          />
          <button>Add Trip</button>
        </form>
      </div>
    );
  };
};

export default TripForm;

// <DateRangePicker
//   startDate={}
//   endDate={}
//   onDatesChange={}
//   focusedInput={}
//   onFocusChange={}
// />