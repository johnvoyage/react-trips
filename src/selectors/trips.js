export const getVisibleTrips = (trips, { text, sortBy, startDate, endDate }) => {
  return trips.filter((trip) => {
    const startDateMatch = typeof startDate !== 'number' || trip.createdAt >= startDate;
    const endDateMatch = typeof endDate !== 'number' || trip.createdAt <= endDate;
    const textMatch = trip.country.toLowerCase().includes(text.toLowerCase());

    return startDateMatch && endDateMatch && textMatch;
  }).sort((a, b) => {
    if (sortBy === 'date') {
      return a.createdAt < b.createdAt ? 1 : -1;
    } else {
      return 1;
    }
    //else if (sortBy === 'amount') {
    //   return a.amount < b.amount ? 1 : -1;
    // }
  });
};

