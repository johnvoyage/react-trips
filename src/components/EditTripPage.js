import React from 'react';
import { connect } from 'react-redux';

import TripForm from './TripForm';
import { editTrip, removeTrip } from '../actions/trips';

const EditTripPage = (props) => (
  <div>
    <TripForm
      trip={props.trip} 
      onSubmit={(trip) => {
        props.dispatch(editTrip(props.trip.id, trip));
        props.history.push('/trips');
      }}
    />
    <button onClick={() => {
      props.dispatch(removeTrip({ id: props.trip.id }));
      props.history.push('/trips');
    }}>Remove</button>
  </div>
);

const mapStateToProps = (state, props) => {
  return {
    trip: state.trips.find((trip) => trip.id === props.match.params.id)
  };
};

export default connect(mapStateToProps)(EditTripPage);