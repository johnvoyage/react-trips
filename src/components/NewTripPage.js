import React from 'react';
import { connect } from 'react-redux';

import TripForm from './TripForm';
import { addTrip } from '../actions/trips';

export class NewTripPage extends React.Component {
  onSubmit = (trip) => {
    // props.dispatch(addTrip(trip));
    this.props.addTrip(trip)
    this.props.history.push('/trips');
  };

  render() {
    return (
      <div> 
        <h1>New Trip</h1>
        <TripForm 
          onSubmit={this.onSubmit}
        />
      </div>
    )
  }
};

const mapDispatchToProps = (dispatch) => ({
    addTrip: (trip) => dispatch(addTrip(trip))
});

export default connect(undefined, mapDispatchToProps)(NewTripPage);