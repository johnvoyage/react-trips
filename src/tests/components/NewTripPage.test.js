import React from 'react';
import { shallow } from 'enzyme';

import { NewTripPage } from '../../components/NewTripPage';
import trips from '../fixtures/trips';


let addTripSpy, historySpy, wrapper;

beforeEach(() => {
  addTripSpy = jest.fn();
  historySpy = { push: jest.fn() };
  wrapper = shallow(<NewTripPage  
    addTrip={addTripSpy}
    history={historySpy}
  />);
});

test('should render new trip page correctly', () => {
  expect(wrapper).toMatchSnapshot();
});

test('should handle onSubmit', () => {
  wrapper.find('TripForm').prop('onSubmit')(trips[1]);

  expect(historySpy.push).toHaveBeenLastCalledWith('/trips');
  expect(addTripSpy).toHaveBeenLastCalledWith(trips[1]);
});