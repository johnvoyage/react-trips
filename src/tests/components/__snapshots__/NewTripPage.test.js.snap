// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render new trip page correctly 1`] = `
<div>
  <h1>
    New Trip
  </h1>
  <TripForm
    onSubmit={[Function]}
  />
</div>
`;
